### PR TITLE
fix: binding context vm when using function without parentheses 

### PR DESCRIPTION
**Pull request by pikax**
Binding function to the vm context.

Fixes: #144

### CODE REVIEW

File: test/setup.spec.js [hunk 1]
```diff
@@ -355,4 +355,46 @@ describe('setup', () => {
       })
       .then(done);
   });
+
+  describe('Bindings', ()=>{
```
Comment (posva) [inline test/setup.spec.js]: If you really want to group them, something like `Methods` would be better
```suggestion
  describe('Methods', ()=>{
```

File: test/setup.spec.js [hunk 2]
```diff
@@ -355,4 +355,46 @@ describe('setup', () => {
...
+    it('should bind the vm when calling with parentesis', async ()=>{
```
Comment (posva) [inline test/setup.spec.js]:
```suggestion
    it('binds methods when calling with parenthesis', async ()=>{
```

File: test/setup.spec.js [hunk 3]
```diff
@@ -355,4 +355,46 @@ describe('setup', () => {
...
+          }
+        }
+      }).$mount();
+  
+      await vm.$el.querySelector('button').click();
+      expect(contextFunction).toBeCalled();
+      expect(context.$el).toBe(vm.$el);
+    });
+
+    it('should bind the vm when calling without parentesis', async ()=>{
```
Comment (posva) [inline test/setup.spec.js]:
```suggestion
    it('binds methods when calling without parenthesis', async ()=>{
```

File: test/setup.spec.js [hunk 4]
```diff
@@ -355,4 +355,46 @@ describe('setup', () => {
...
+        setup() {
+          return {
+            contextFunction
+          }
+        }
+      }).$mount();
+  
+      await vm.$el.querySelector('button').click();
+      expect(contextFunction).toBeCalled();
+      expect(context.$el).toBe(vm.$el);
```
Comment (posva) [inline test/setup.spec.js]: this could be moved inside of the `contextFunction` and get rid of the context variable

File: test/setup.spec.js [hunk 5]
```diff
@@ -355,4 +355,46 @@ describe('setup', () => {
...
```
Comment (posva) [inline test/setup.spec.js]: same

File: test/setup.spec.js (hunk 4 above)
Comment (posva) [reply test/setup.spec.js]: also, shouldn't we just compare to `vm` directly?

Comment (pikax) [reply test/setup.spec.js]: when I created the tests `vm` was being proxied, but now because of the `bind` I can do that 👍

**Comment by anhoev** [comment]
Please merge, i need it because context.slots is not reactive in runtime

**Pull request merged**